webapp/src/containers/SubscribeModal/index.tsx

```diff
@@ -0,0 +1,234 @@
+        });
+        dispatch(toggleShowSubscribeModal({isVisible: false, commandArgs: []}));
+    };
+
+    // Set organization name
+    const onOrganizationChange = (value: string) => {
+        setErrorState({...errorState, organization: ''});
+        setSubscriptionDetails({...subscriptionDetails, organization: value});
+    };
+
+    // Set project name
+    const onProjectChange = (value: string) => {
+        setErrorState({...errorState, project: ''});
+        setSubscriptionDetails({...subscriptionDetails, project: value});
+    };
+
+    // Set event type
+    const onEventTypeChange = (value: string) => {
+        setErrorState({...errorState, eventType: ''});
+        setSubscriptionDetails({...subscriptionDetails, eventType: value});
+    };
+
+    // Set channel name
+    const onChannelChange = (value: string) => {
+        setErrorState({...errorState, channelID: ''});
+        setSubscriptionDetails({...subscriptionDetails, channelID: value});
+    };
+
+    // Handles on confirming subscription
+    const onConfirm = () => {
+        const errorStateChanges: SubscriptionPayload = {
+            organization: '',
+            project: '',
+            channelID: '',
+            eventType: '',
+        };
+        if (subscriptionDetails.organization === '') {
+            errorStateChanges.organization = 'Organization is required';
+        }
+        if (subscriptionDetails.project === '') {
+            errorStateChanges.project = 'Project is required';
+        }
+        if (subscriptionDetails.eventType === '') {
+            errorStateChanges.eventType = 'Event type is required';
+        }
+        if (subscriptionDetails.channelID === '') {
+            errorStateChanges.channelID = 'Channel name is required';
+        }
+        if (errorStateChanges.organization || errorStateChanges.project || errorStateChanges.channelID || errorStateChanges.eventType) {
+            setErrorState(errorStateChanges);
+            return;
+        }
+
+        // Make POST api request
+        createSubscription();
+    };
+
+    // Make POST api request to create subscription
+    const createSubscription = async () => {
+        const createSubscriptionResponse = await usePlugin.makeApiRequest(plugin_constants.pluginApiServiceConfigs.createSubscription.apiServiceName, subscriptionDetails);
+        if (createSubscriptionResponse) {
+            onHide();
+        }
+    };
+
+    useEffect(() => {
+        const channelList = getChannelState().data;
+        if (channelList) {
+            setChannelOptions(channelList.map((channel) => ({label: <span><i className='fa fa-globe dropdown-option-icon'/>{channel.display_name}</span>, value: channel.id})));
+        }
+
+        const projectList = getProjectState().data;
+        if (projectList) {
+            setProjectOptions(getProjectList(projectList));
+            setOrganizationOptions(getOrganizationList(projectList));
+        }
+    }, [usePlugin.state]);
```

Comment:
create required selector and use specific state change params here instead of entire state